luigi/worker.py

```diff
@@ -490,9 +556,16 @@ def _get_work(self):
 
     def _run_task(self, task_id):
         task = self._scheduled_tasks[task_id]
-        p = TaskProcess(task, self._id, self._task_result_queue,
-                        random_seed=bool(self.worker_processes > 1),
-                        worker_timeout=self.__worker_timeout)
+
+        if task.run == NotImplemented:
+            p = ExternalTaskProcess(task, self._id, self._task_result_queue,
+                                    random_seed=bool(self.worker_processes > 1),
+                                    worker_timeout=self.__worker_timeout)
```

Comment:
Trying to understand your code. Doesn't this mean it will only be run once, possibly at the beginning of the pipeline? What you really want to do is re-check the existence later.